cms/wp/uploads/pull.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * Wordpress Uploads Pull (rsync)
+ *
+ * @file
+ * @package wp-deployer
+ */
+
+namespace Deployer;
+
+task(
+	'wp:uploads:pull',
+	function() {
+		$upload_dir = 'web/app/uploads';
```

Comment:
Should this path be configurable?

Reply:
This is more of a wordpress bedrock issue.  You could also configure a symlink in the ansible scripts but Wordpress does expect it there 'web/app/uploads'.

Reply:
I guess my point is that not every WP instance will be using that exact path.  For example, some might name their web root `web` or `public` or something else.  And, as you noted, not everyone will be using Bedrock's directory layout.

If you look at some of UT's Deployer 6 recipes you'll see that we'd typically create a variable named `web_root` to point to that `web` or `public` folder:

```php
set('web_root', '{{release_path}}/web');
```

And then another variable pointing to the upload directory:

```php
set('uploads', '{{web_root}}/app/uploads');
```

This allowed us to create a reusable recipe where developers only needed to change, at most, one or two variables and all tasks using those vars "just worked".  I'd propose using a similar approach here so that these `wp` tasks can be used by anyone regardless of how their repo is set up.

Reply:
While your approach works with Deployer 6 and Drupal, have you actually tested it on Deployer 7 and Wordpress?  I'd love to assume that Wordpress is altruistic enough to not hard code the app/uploads directory name, but I wouldn't force that (variable dir name)  on others unless we prove it can be done.

Reply:
I can confirm that Deployer 7 does support configuration variables and string interpolation virtually the same way as Deployer 6, which we have used with WordPress in the past.  Their [default recipes](https://github.com/deployphp/deployer/tree/v7.0.0-beta.25/recipe) follow the pattern I described of using configuration to set a sane default path while allowing users of that recipe to override it as needed.

For example, in [their WordPress recipe](https://github.com/deployphp/deployer/blob/v7.0.0-beta.25/recipe/wordpress.php), they're using the `shared_dirs` variable to inform the `deploy:shared` task of what paths need to be symlinked.

Another example is the [Symfony recipe](https://github.com/deployphp/deployer/blob/v7.0.0-beta.25/recipe/symfony.php) where the `bin/console` config variable is set to the typical place that script lives, but because it's a config setting it can be tweaked by any user of that recipe.

I suggest we follow the same pattern in our recipes where paths are set in configuration (with a sane default value of your preference) so they can be tweaked if needed.

Reply:
done.